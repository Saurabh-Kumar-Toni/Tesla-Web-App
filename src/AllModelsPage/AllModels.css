.AllCarContainer{

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100%;
}

.cardlist{
    display: flex;
    flex-direction: row;
    margin-right: 2px;
    justify-content: space-between;
}

@media only screen  and (max-width : 376px){
   
    .cardlist{
        flex-direction: column;
        flex-wrap: nowrap;
        overflow: scroll;
        margin-top: 50px;
    }
    
}