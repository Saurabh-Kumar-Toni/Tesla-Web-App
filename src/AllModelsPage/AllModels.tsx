import React, { useEffect, useState } from 'react';
import Card from '../Components/Card';
import './AllModels.css'

const AllModels:React.FC = () => {

    const [modelData, setModelData] = useState([]);
    useEffect(() => {
        fetch('https://tesla-app-server.herokuapp.com/models/all')
        .then((res) => {
            return res.json();
        }).then((data) => setModelData(data.elements))
    },[])

    const renderModelBody = () => {
        const list = modelData.map(({displayName,model,topSpeed,range,peakPower,acceleration, AWD}, id=0) => {
            return(
                <div key={id}>
                    <Card
                        displayName={displayName}
                        model={model}
                        topSpeed={topSpeed}
                        range={range}
                        peakPower={peakPower}
                        acceleration={acceleration}
                        AWD={AWD}
                    />
                </div>
            )
        })
        return list;
    }

    return (
        <div className="AllCarContainer">
            <div className={'cardlist'}>
            {modelData.length> 0 ? renderModelBody():""}
            </div>
        </div>
    )
}

export default AllModels;