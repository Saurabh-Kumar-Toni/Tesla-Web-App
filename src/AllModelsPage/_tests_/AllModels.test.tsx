import { render, screen } from '@testing-library/react';
import AllModels from '../AllModels';

test('check if the All Cars Page is rendered properly or not', () => {
 
    const {container} = render(<AllModels />)
    expect(container.firstChild).toMatchSnapshot();
  });