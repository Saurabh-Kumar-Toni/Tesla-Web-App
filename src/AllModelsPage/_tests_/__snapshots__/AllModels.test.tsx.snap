// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if the All Cars Page is rendered properly or not 1`] = `
<div
  class="AllCarContainer"
>
  <div
    class="cardlist"
  />
</div>
`;
